versionutils/version.go

```diff
@@ -133,6 +133,33 @@ func (v Version) IsGreaterThan(lesser Version) (bool, bool) {
 	return false, true
 }
 
+// In order, returns isGreaterThanOrEqualTo, isDeterminable
+// labelOrder specifies tie-break order for labels
+// e.g. labelOrder = [ beta, alpha, predev ], then 1.7.0-beta11 > 1.7.0-alpha5 > 1.7.0-predev9
+// isDeterminable is for incomporable versions because they have different labels not specified in labelOrder
+func (v Version) IsGreaterThanWithLabelOrder(lesser Version, labelOrder []string) (bool, bool) {
+
+	greaterThan, determinable := v.IsGreaterThan(lesser)
+	if determinable {
+		return greaterThan, determinable
+	}
+	vIndex, lIndex := -1, -1
+	for i, lbl := range labelOrder {
+		if lbl == v.Label {
+			vIndex = i
+		}
+		if lbl == lesser.Label {
+			lIndex = i
+		}
+	}
+	if vIndex == -1 || lIndex == -1 {
+		return false, false
```

Comment:
this should be an error instead?

Reply:
I would argue that setting `isDeterminable` to `false` works logically, rather than having to propagate an error all the way up, similar to what was being done already.

Reply:
I can get on board with that 👍